docker/developers/install-local-nvm.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+set -e
+
+# This script works alongside the assumptions make in the Dockerfile to support
+# updating a containerized install of NVM/Node/npm for use within the container.
+
+if [ "$(hostname)" != 'install-nvm' ]; then
+	echo "This is meant to run inside the docker compose context."
+	echo "Run with: 'docker compose run install-nvm'"
+	exit 1
+fi
+
+echo "Installing nvm into the Docker 'nvm' volume."
+
+# Copy over a pre-built copy of nvm
+if [ ! -e ${HOME}/.nvm/package.json ]; then
+  cp -r /opt/base-nvm/* ${HOME}/.nvm/.
+fi
+
+# Initialize NVM
+. ${HOME}/.nvm/nvm.sh
+
+# Determine the desired node version
+NODE_VERSION=`jq -r .engines.node < apps/package.json | sed "s/[\^]//g"`
+echo " - Installing node v${NODE_VERSION}"
+nvm install ${NODE_VERSION}
+nvm alias default ${NODE_VERSION}
+nvm use default
+
+# Install the correct version of NPM
+NPM_VERSION=`jq -r .engines.npm < apps/package.json | sed "s/[\^]//g"`
```

Comment:
I believe we can preserve the semver syntax, otherwise this would pin us to an older version when we want the latest minor on the same major.

```suggestion
NPM_VERSION=`jq -r .engines.npm < apps/package.json`
```

Reply:
I want every developer to have the same version of npm! Ideally, the same version as production. How to make this happen?